### PR TITLE
[specs][kimchi] generate list of alpha powers from code

### PR DESCRIPTION
**Pull request by mimoo**
This might be a bit too much...

### CODE REVIEW

File: book/specifications/kimchi/template.md [hunk 1]
```diff
@@ -56,6 +56,10 @@ TODO: use expr to define the index columns?
 
 ### Gates
 
+<!-- generated using `cargo run --bin code-extractor` -->
```
Comment (querolita) [inline book/specifications/kimchi/template.md:59]: I like this clear layout

File: book/specifications/code-extractor/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[package]
```
Comment (querolita) [inline book/specifications/code-extractor/Cargo.toml:1]: does cargo really need a whole new library for this?

File: kimchi/src/alphas.rs [hunk 3]
```diff
@@ -171,6 +172,33 @@ impl<F: Field> Alphas<F> {
...
+            } else {
+                format!("{:?}", arg)
+            };
+            let range = self
+                .mapping
+                .get(&arg)
+                .expect("you need to register all arguments before calling display");
+            writeln!(
+                f,
+                "* **{}**. Offset starts at {} and {} powers of $\\alpha$ are used",
```
Comment (querolita) [inline kimchi/src/alphas.rs:193]: Very nice display

**Review by querolita — COMMENTED** [review]
I like the explicit display of the alpha powers, that could change dynamically if the offsets ever change, but I am not sure I love the idea of the new library code-extractor only being used for this task...

**Comment by mimoo** [comment]
yeah I'm not really happy with the library thing. I'm wondering what's a good alternative.

**Comment by mimoo** [comment]
let me remove the library part, I can just create a test to get that

**Comment by mimoo** [comment]
@querolita what do you think now? the test will panic with a message telling you to copy the paragraph to the spec:

```
running 1 test
thread 'alphas::tests::get_alphas_for_spec' panicked at 'the specification of kimchi must contain the following paragraph:

* **gates**. Offset starts at 0 and 21 powers of $\alpha$ are used
* **Permutation**. Offset starts at 21 and 3 powers of $\alpha$ are used

```

File: kimchi/src/alphas.rs [hunk 4]
```diff
@@ -280,4 +310,39 @@ mod tests {
...
+    // useful for the spec
+
+    use crate::{
+        circuits::{gate::CircuitGate, wires::Wire},
+        linearization::expr_linearization,
+        prover_index::testing::new_index_for_test,
+    };
+
+    #[test]
+    fn get_alphas_for_spec() {
```
Comment (querolita) [inline kimchi/src/alphas.rs:323]: Cool, I find this way cleaner

**Review by querolita — APPROVED** [review]

**Pull request merged**